### PR TITLE
Alphabetize jsdoc comments and methods in classes

### PR DESCRIPTION
**Pull request by jtsshieh**
also added oldMessage.pinned to typings. idk if it is supposed to be null or undefined though.

### CODE REVIEW

**Comment by jtsshieh** [comment]
if you can, please merge my pr into this your fork because I updated oldChannel and oldRole. Don't want any conflicts there. (or all the latest dev branch changes).

**Comment by jtsshieh** [comment]
Client still needs to be "alphabetized"

File: index.d.ts [hunk 1]
```diff
@@ -1212,6 +1217,7 @@ declare namespace Eris {
     ): Promise<Webhook>;
     createGroupChannel(userIDs: string[]): Promise<GroupChannel>;
     createGuild(name: string, options?: CreateGuildOptions): Promise<Guild>;
+    // EmojiOptions has options.icon. Supposed to be options.image
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
```

File: lib/structures/Guild.js [hunk 2]
```diff
@@ -678,37 +663,33 @@ class Guild extends Base {
...
+    * Begin pruning the guild
+    * @arg {Number} [options] The options to pass to prune members
+    * @arg {Number} [options.days=7] The number of days of inactivity to prune for
+    * @arg {Array<String>} [options.includeRoles] An array of role IDs that members must have to be considered for pruning
+    * @arg {String} [options.reason] The reason to be displayed in audit logs
+    * @returns {Promise<Number>} Resolves with the number of pruned members
     */
-    getBan(userID) {
-        return this._client.getGuildBan.call(this._client, this.id, userID);
+    pruneMembers(options) {
```
Comment (bsian03) [inline lib/structures/Guild.js:707]: Missing `computePruneCount`

**Comment by jtsshieh** [comment]
You should probably wait because i'm resolving the merge conflicts and that might change things

File: index.d.ts [hunk 3]
```diff
@@ -1373,6 +1380,7 @@ declare namespace Eris {
     leaveGuild(guildID: string): Promise<void>;
     leaveVoiceChannel(channelID: string): void;
     pinMessage(channelID: string, messageID: string): Promise<void>;
+    // Note: PruneMemberOptions is missing `computerPruneCount`
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
```

File: index.d.ts [hunk 4]
```diff
@@ -1683,6 +1687,8 @@ declare namespace Eris {
     splash: string | null;
     splashURL: string | null;
     constructor(data: BaseData, client: Client);
+    dynamicIconURL(format?: string, size?: number): string;
+    dynamicSplashURL(format?: string, size?: number): string;
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
    dynamicIconURL(format?: ImageFormat, size?: number): string;
    dynamicSplashURL(format?: ImageFormat, size?: number): string;
```

File: lib/structures/Guild.js (hunk 2 above)
Comment (jtsshieh) [reply lib/structures/Guild.js:707]: do you mean getPruneCount?

Comment (bsian03) [reply lib/structures/Guild.js:707]: nope, this is supposed to be consistent with pruneMembers in client class

File: lib/Client.js [hunk 5]
```diff
@@ -49,77 +49,80 @@ const sleep = (ms) => new Promise((res) => setTimeout(res, ms));
...
+* @prop {Object} guildShardMap Object mapping guild IDs to shard IDs
+* @prop {Object} notes Object mapping user IDs to user notes (user accounts only)
+* @prop {Object} options Eris options
 * @prop {Object} privateChannelMap Object mapping user IDs to private channel IDs
 * @prop {Collection<PrivateChannel>} privateChannels Collection of private channels the bot is in
-* @prop {Collection<GroupChannel>} groupChannels Collection of group channels the bot is in (user accounts only)
-* @prop {Collection<VoiceConnection>} voiceConnections Extended collection of active VoiceConnections the bot has
-* @prop {Object} guildShardMap Object mapping guild IDs to shard IDs
+* @prop {Collection<Relationship>} relationships Collection of relationships the bot user has (user accounts only)
+// Request handler?
```
Comment (bsian03) [inline lib/Client.js:63]: Yeah I've already done docs for TS for Req Handler and related

File: lib/structures/Guild.js (hunk 2 above)
Comment (jtsshieh) [reply lib/structures/Guild.js:707]: Don't think it was there originally. I added it in

File: index.d.ts [hunk 6]
```diff
@@ -1683,7 +1690,6 @@ declare namespace Eris {
     position: number;
     type: Exclude<ChannelTypes, 1 | 3>;
     constructor(data: BaseData, guild: Guild);
-    createInvite(options?: CreateInviteOptions, reason?: string): Promise<Invite & InviteWithoutMetadata<null>>;
```
Comment (bsian03) [inline index.d.ts:1686]: This needs to be added back

File: index.d.ts [hunk 7]
```diff
@@ -1857,7 +1863,6 @@ declare namespace Eris {
   }
 
   export class PermissionOverwrite extends Permission {
-    createdAt: number;
```
Comment (bsian03) [inline index.d.ts]: Add back

File: lib/Client.js (hunk 5 above)
Comment (bsian03) [inline lib/Client.js]: Doc needed

File: lib/Client.js [hunk 8]
```diff
@@ -49,77 +49,80 @@ const sleep = (ms) => new Promise((res) => setTimeout(res, ms));
 /**
 * Represents the main Eris client
 * @extends EventEmitter
-* @prop {String} token The bot user token
 * @prop {Boolean?} bot Whether the bot user belongs to an OAuth2 application
-* @prop {Object} options Eris options
 * @prop {Object} channelGuildMap Object mapping channel IDs to guild IDs
-* @prop {Collection<Shard>} shards Collection of shards Eris is using
+// Gateway URL?
```
Comment (bsian03) [inline lib/Client.js]: Doc needed

File: lib/structures/Message.js [hunk 9]
```diff
@@ -8,37 +8,40 @@ const User = require("./User");
...
 * @prop {String} messageReference.messageID The id of the original message this message was crossposted from
 * @prop {String} messageReference.channelID The id of the channel this message was crossposted from
 * @prop {String} messageReference.guildID The id of the guild this message was crossposted from
-* @prop {Number} flags Message flags (see constants)
-* @prop {Object[]} attachments Array of attachments
-* @prop {Object[]} embeds Array of embeds
-* @prop {Object?} activity The activity specified in the message
-* @prop {Object?} application The application of the activity in the message
+* @prop {Boolean} pinned Whether the message is pinned or not
+// Note add prefix?
```
Comment (bsian03) [inline lib/structures/Message.js]: Doc needed

File: index.d.ts (hunk 6 above)
Comment (jtsshieh) [reply index.d.ts:1686]: GuildChannel doesn't have createinvite, only textchannel does 🤔

File: index.d.ts (hunk 7 above)
Comment (jtsshieh) [reply index.d.ts]: done

File: lib/structures/Message.js (hunk 9 above)
Comment (jtsshieh) [reply lib/structures/Message.js]: done

File: lib/Client.js (hunk 8 above)
Comment (jtsshieh) [reply lib/Client.js]: done

File: lib/Client.js (hunk 5 above)
Comment (jtsshieh) [reply lib/Client.js]: done

File: index.d.ts (hunk 6 above)
Comment (jtsshieh) [reply index.d.ts:1686]: Or is something i'm missing

Comment (bsian03) [reply index.d.ts:1686]: Yup that's my bad

File: index.d.ts [hunk 10]
```diff
@@ -1306,6 +1311,7 @@ declare namespace Eris {
     ): Promise<Emoji>;
     editGuildIntegration(guildID: string, integrationID: string, options: IntegrationOptions): Promise<void>;
     editGuildMember(guildID: string, memberID: string, options: MemberOptions, reason?: string): Promise<void>;
+    editGuildWidget(guildID: string, options: Widget) : Promise<Widget>
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
    editGuildWidget(guildID: string, options: Widget): Promise<Widget>
```

File: index.d.ts [hunk 11]
```diff
@@ -1348,6 +1354,7 @@ declare namespace Eris {
     getGuildPreview(guildID: string): Promise<GuildPreview>;
     getGuildVanity(guildID: string): Promise<{ code?: string; uses?: number }>;
     getGuildWebhooks(guildID: string): Promise<Webhook[]>;
+    getGuildWidget(guildID: string) : Promise<Widget>;
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
    getGuildWidget(guildID: string): Promise<Widget>;
```

**Review by bsian03 — CHANGES_REQUESTED** [review]

**Review by bsian03 — APPROVED** [review]

**Pull request merged**